fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/OptimizeExpressionTask.java

```diff
@@ -11,49 +11,64 @@
- * transformations and applying those rules.
+ * transformations and applying those rules, and to drive group's statistics.
  * <p>
- * If exploreOnly is false:
+ * If isExplore is false:
  * OptimizeExpressionTask optimizes a GroupExpression by constructing all logical
- * and physical transformations and applying those rules.
+ * and physical transformations and applying those rules, and to drive group's statistics.
  * <p>
  * The rules are sorted by promise and applied in that order
  * so a physical transformation rule is applied before a logical transformation rule.
  */
 public class OptimizeExpressionTask extends OptimizerTask {
     private final GroupExpression groupExpression;
+    private final boolean isExplore;
 
     OptimizeExpressionTask(TaskContext context, GroupExpression expression) {
         super(context);
         this.groupExpression = expression;
+        this.isExplore = false;
+    }
+
+    OptimizeExpressionTask(TaskContext context, GroupExpression expression, boolean isExplore) {
+        super(context);
+        this.groupExpression = expression;
+        this.isExplore = isExplore;
     }
 
     private List<Rule> getValidRules() {
         List<Rule> validRules = Lists.newArrayListWithCapacity(RuleType.NUM_RULES.id());
-        List<Rule> logicalRules = context.getOptimizerContext().getRuleSet().
-                getTransformRules();
+        List<Rule> logicalRules = context.getOptimizerContext().getRuleSet().getTransformRules();
         filterInValidRules(groupExpression, logicalRules, validRules);
 
-        List<Rule> physicalRules = context.getOptimizerContext().getRuleSet().
-                getImplementRules();
-        filterInValidRules(groupExpression, physicalRules, validRules);
+        if (!isExplore) {
+            List<Rule> physicalRules = context.getOptimizerContext().getRuleSet().getImplementRules();
+            filterInValidRules(groupExpression, physicalRules, validRules);
+        }
+
+        validRules.sort(Comparator.comparingInt(Rule::promise));
         return validRules;
     }
 
     @Override
     public String toString() {
-        return "OptimizeExpressionTask for groupExpression " + groupExpression;
+        return "OptimizeExpressionTask" + (this.isExplore ? "[explore]" : "")
+                + " for groupExpression " + groupExpression;
     }
 
     @Override
     public void execute() {
         List<Rule> rules = getValidRules();
-        rules.sort(Comparator.comparingInt(Rule::promise));
 
         for (Rule rule : rules) {
-            pushTask(new ApplyRuleTask(context, groupExpression, rule));
+            pushTask(new ApplyRuleTask(context, groupExpression, rule, isExplore));
+        }
+
+        pushTask(new DeriveStatsTask(context, groupExpression));
+        for (int i = groupExpression.arity() - 1; i >= 0; i--) {
+            pushTask(new ExploreGroupTask(context, groupExpression.getInputs().get(i)));
```

Comment:
Is this a time-consuming place? We will explore the groupExpression and then derive their statistics?

Reply:
yeah, explore all child group will increase the time, I will run performance later. we explore all groupExpression for get an accurate statistics